pilot/pkg/networking/plugin/authz/rbac_test.go

```diff
@@ -545,6 +572,48 @@ func TestConvertRbacRulesToFilterConfig(t *testing.T) {
 		}},
 	}
 
+	policy3 := &policy.Policy{
+		Permissions: []*policy.Permission{{
+			Rule: &policy.Permission_AndRules{
+				AndRules: &policy.Permission_Set{
+					Rules: []*policy.Permission{
+						{
+							Rule: &policy.Permission_OrRules{
+								OrRules: &policy.Permission_Set{
+									Rules: []*policy.Permission{
+										{
+											Rule: &policy.Permission_Header{
```

Comment:
Shall we add some utility function here? This is hard to read.

Reply:
Done.

Reply:
Can we make it more general, and reuse it for any other claims (instead of just groups claim)?

Reply:
The original tests in this file contain various nested testing permissions and claims (the following is an example), which are not straightforward to be made general. To make them all general requires an overhaul of the original code. As an overhaul of the existing code is not directly related to this PR, I suggest doing it in a different PR.

					Rules: []*policy.Permission{
						{
							Rule: &policy.Permission_OrRules{
								OrRules: &policy.Permission_Set{
									Rules: []*policy.Permission{
										{
											Rule: &policy.Permission_DestinationPort{
												DestinationPort: 80,
											},
										},
										{
											Rule: &policy.Permission_DestinationPort{
												DestinationPort: 443,
											},
										},
									},
								},
							},
						},
						{
							Rule: &policy.Permission_OrRules{
								OrRules: &policy.Permission_Set{
									Rules: []*policy.Permission{
										{
											Rule: &policy.Permission_DestinationIp{
												DestinationIp: &core.CidrRange{
													AddressPrefix: "192.1.2.0",
													PrefixLen:     &types.UInt32Value{Value: 24},
												},
											},
										},
										{
											Rule: &policy.Permission_DestinationIp{
												DestinationIp: &core.CidrRange{
													AddressPrefix: "2001:db8::",
													PrefixLen:     &types.UInt32Value{Value: 28},
												},
											}},
									},
								},
							},
						},
						{
							Rule: &policy.Permission_OrRules{
								OrRules: &policy.Permission_Set{
									Rules: []*policy.Permission{
										{
											Rule: &policy.Permission_Header{
												Header: &route.HeaderMatcher{
													Name: "key1",
													HeaderMatchSpecifier: &route.HeaderMatcher_PrefixMatch{
														PrefixMatch: "prefix",
													},
												},
											},
										},
										{
											Rule: &policy.Permission_Header{
												Header: &route.HeaderMatcher{
													Name: "key1",
													HeaderMatchSpecifier: &route.HeaderMatcher_SuffixMatch{
														SuffixMatch: "suffix",
													},
												},
											},
										},
									},
								},
							},
						},
						{
							Rule: &policy.Permission_OrRules{
								OrRules: &policy.Permission_Set{
									Rules: []*policy.Permission{
										{
											Rule: &policy.Permission_Header{
												Header: &route.HeaderMatcher{
													Name: "key2",
													HeaderMatchSpecifier: &route.HeaderMatcher_ExactMatch{
														ExactMatch: "simple",
													},
												},
											},
										},
										{
											Rule: &policy.Permission_Header{
												Header: &route.HeaderMatcher{
													Name: "key2",
													HeaderMatchSpecifier: &route.HeaderMatcher_PresentMatch{
														PresentMatch: true,
													},
												},
											},
										},
									},
								},
							},
						},
					},
				},
			},
		}},

Reply:
Ok. I am fine to address it in a separate PR.